independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/RuntimeExceptionMapper.java

```diff
@@ -226,27 +233,75 @@ private Map<Class<? extends Throwable>, ResourceExceptionMapper<? extends Throwa
         return context.getTarget() != null ? context.getTarget().getClassExceptionMappers() : null;
     }
 
-    @SuppressWarnings({ "unchecked", "rawtypes" })
     private <T extends Throwable> Map.Entry<Throwable, jakarta.ws.rs.ext.ExceptionMapper<? extends Throwable>> doGetExceptionMapper(
             Class<T> clazz,
             Map<Class<? extends Throwable>, ResourceExceptionMapper<? extends Throwable>> mappers,
             Throwable throwable) {
+        Stream<Supplier<AbstractMap.Entry<Throwable, ExceptionMapper<? extends Throwable>>>> mappingStrategies = Stream.of(
+                // Check if the exception should always be unwrapped (if strategy=ALWAYS).
+                // In this case, search mapping for the wrapped exception, ignoring class hierarchy
+                () -> searchMapperForExceptionsToUnwrap(clazz, mappers, throwable, ExceptionUnwrapStrategy.ALWAYS),
+                // If the exception type is directly mapped, ignore the unwrapping.
+                () -> buildMapperEntryIfExists(clazz, mappers, throwable),
+                // Check if the exception should be unwrapped if no exact match (if strategy=UNWRAP_IF_NO_EXACT_MATCH).
+                () -> searchMapperForExceptionsToUnwrap(clazz, mappers, throwable,
+                        ExceptionUnwrapStrategy.UNWRAP_IF_NO_EXACT_MATCH),
+                // Walk up the class hierarchy looking for a mapper for the type
+                () -> searchMapperInClassHierarchy(clazz, mappers, throwable),
+                // If no mapper found and exception is marked for unwrapping (if strategy=UNWRAP_IF_NO_MATCH), unwrap it
+                () -> searchMapperForExceptionsToUnwrap(clazz, mappers, throwable, ExceptionUnwrapStrategy.UNWRAP_IF_NO_MATCH));
```

Comment:
We try hard to avoid lambdas in code that executes at runtime. Can you please update the code to remove them?